extensions/reactive-db2-client/deployment/src/main/java/io/quarkus/reactive/db2/client/deployment/ReactiveDB2ClientProcessor.java

```diff
@@ -82,8 +93,30 @@ ServiceStartBuildItem build(BuildProducer<FeatureBuildItem> feature,
     }
 
     @BuildStep
-    HealthBuildItem addHealthCheck(DataSourcesBuildTimeConfig dataSourcesBuildTimeConfig) {
-        return new HealthBuildItem("io.quarkus.reactive.db2.client.runtime.health.ReactiveDB2DataSourceHealthCheck",
-                dataSourcesBuildTimeConfig.healthEnabled);
+    void addHealthCheck(
+            BuildProducer<HealthBuildItem> healthChecks,
+            DataSourcesBuildTimeConfig dataSourcesBuildTimeConfig,
+            DataSourceReactiveBuildTimeConfig dataSourceReactiveBuildTimeConfig) {
+        if (!createPool(dataSourcesBuildTimeConfig, dataSourceReactiveBuildTimeConfig)) {
+            return;
+        }
+
+        healthChecks
+                .produce(new HealthBuildItem("io.quarkus.reactive.db2.client.runtime.health.ReactiveDB2DataSourceHealthCheck",
+                        dataSourcesBuildTimeConfig.healthEnabled));
+    }
+
+    private static boolean createPool(DataSourcesBuildTimeConfig dataSourcesBuildTimeConfig,
```

Comment:
Can we have a comment on what the rationale is behind this?

Reply:
It looks simple enough to me? It just checks the config to know if we should create the pool.

Reply:
What I don't get is why, not how it's done. Maybe it's just because I have no knowledge of this part of the code. Feel free to ignore

Reply:
Yeah. The why is that you don't want to create the pool if it's not configured or targets another database type than the one supported by this extension :).

Keep in mind that the idea is to support multiple datasources here too at some point so we need to support the case where we have multiple reactive client extensions at the same time.

It's probably going to be more obvious once we add support for multiple datasources there too.

I wouldn't mind an approval so that I can get this one in and fix @rsvoboda 's issue :).